Stats: Fix for full-width buttons in purchase flows

a8ck3n opened the pull request:
Also updates buttons on both flows (commercial & pwyw) to use the same CSS for full-width buttons.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to:

https://github.com/Automattic/red-team/issues/44

## Proposed Changes

Fixes the full-width buttons to work correctly in Odyssey Stats. Was previously working in Calypso but not correctly aligning the text in Odyssey. Should look like this now:

#### PWYW flow

<img width="372" alt="SCR-20240528-rkwh" src="https://github.com/Automattic/wp-calypso/assets/40267301/1d315db3-b8d6-4b80-9417-f0a57e6a7aed">

#### Commercial flow

<img width="393" alt="SCR-20240528-rkyw" src="https://github.com/Automattic/wp-calypso/assets/40267301/36482f28-52af-4596-aba6-29dd90ae89b0">

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Buttons were not drawn correctly in Odyssey Stats.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Follow steps listed [here](PCYsg-Pp7-p2) to test these changes in Odyssey Stats.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

a8ck3n committed:
Fix for full-width buttons.

Also updates buttons on both flows (commercial & pwyw) to use the same CSS for full-width buttons.

kangzj commented on client/my-sites/stats/stats-purchase/stats-purchase-personal.tsx:
```diff
@@ -272,7 +272,7 @@ const PersonalPurchase = ( {
 			) : (
 				<div className={ `${ COMPONENT_CLASS_NAME }__actions` }>
 					<ButtonComponent
-						className="stats-purchase-page__full-width"
```
I understand according to BEM rules, modifiers are supposed to look like `is-full-width`, but with the context of Odyssey, it would/have cause issues. CSS naming space is stuffed in Odyssey, unlike Calypso where it's just a standalone app. I would still lean to the original naming on this one if it makes sense. 🙂

grzegorz-cp replied:
Generic CSS classes also tend to leak when navigating between Calypso pages. We have had several issues like this over the past few months.

a8ck3n replied:
Okay, that makes sense. So for cases like this, we'd prefer to use a longer, more specific class name.

kangzj reviewed (approved):
It works as expected. Left a suggestion inline.

Found a styling issue not related to the PR - the circled line seems to be off. Also I didn't find it in the design 🤔 Happy for it to be addressed separately.

![image](https://github.com/Automattic/wp-calypso/assets/1425433/768a0f0a-0d3c-4441-a611-059c9b947faf)

Thanks for working on this! ❤️

grzegorz-cp commented on client/my-sites/stats/stats-purchase/styles.scss:
```diff
@@ -545,27 +545,15 @@ $button-padding: 8px 32px;
-
-	button {
-		@media (max-width: $break-small) {
+	.components-button {
```
Nesting multiple class names creates issues when people need to overwrite styling in the future. This makes a selector stronger unnecessarily.

You are not applying any styling properties on lines 550 and 552, so a single class would be a better choice.

a8ck3n replied:
@grzegorz-cp — I was actually having some trouble getting this working as the existing classes where overriding it. I'll give it another look now and shoot you a ping if I need help getting this right.

a8ck3n replied:
The styles I'm trying to override are part of the `.components-button` scope with a specificity of `0-1-0`. I've update my selector to `stats-purchase-wizard__actions.components-button` to get a specificty of `0-2-0`. This matches what I'm seeing in the CSS for the `stats-purchase-single__personal-checklist-button`.

Is this more inline with what you were thinking @grzegorz-cp?

a8ck3n committed:
More descriptive class name for use with Odyssey

a8ck3n committed:
Make “continue with free” button full width.

a8ck3n committed:
Remove unused styles.

a8ck3n committed:
Add vertical margin to notices.

Specfically for the have plan and upgrading flow, otherwise it butts up against the following paragraph text.

a8ck3n committed:
Improve CSS speceficity for full-width buttons
